### DISCUSSION TITLE
🟦 Flying Cube - Copilot Game Challenge 🟥

### DISCUSSION THREAD

**Discussion by morganersery1** [community/community#138419] [GitHub Education]
![GXNj-ipW8AAFoG0](https://github.com/user-attachments/assets/a839ae01-ba39-4967-9cee-6eea622f7b8c)

For the next two weeks (9/11-25) we're running a Copilot challenge where you can take a simple game and add your own twist to it for a chance to win an exclusive artist Mona print!

### Challenge instructions
Challenge start: September 11, 2024, 3pm EST
Challenge end: September 25, 2024, 11:59pm EST

- [Make sure you have the student developer pack](https://gh.io/AAs68cr)
- [Fork this Repo](https://github.com/MetzinAround/CubeFlyer)
- [Clone the repo locally](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository#about-cloning-a-repository)
- Open the code in the editor of your choice
- Use GitHub Copilot to make some fun changes and improvements to the game!

We'll pick our favorite submissions based on which changes we are most entertained/amazed by!

- [Push your cloned repo back to GitHub](https://docs.github.com/en/get-started/using-git/pushing-commits-to-a-remote-repository)
- Share the link to your repo in **this discussion thread** to enter the giveaway!

We will award a maximum of 5 winners with one of these [Mona prints](https://www.instagram.com/p/C8FRMedxd8L/?utm_source=ig_web_copy_link&igsh=MzRlODBiNWFlZA==).

<img width="489" alt="Screenshot 2024-09-11 at 2 03 38 PM" src="https://github.com/user-attachments/assets/860403a5-9096-41a4-bee6-dc8bfb857cff">

Full Terms and Conditions can be found [here](https://github.com/github-education-resources/Student-Developer-Pack-Current-Partners-FAQ/blob/main/Copilot-Game-Challenge-Terms-and-Conditions.md). 

### To Help You Get Started
[Watch our Twitch Livestream with a Copilot PM](https://www.twitch.tv/videos/2248549317?collection=0UAVKpQ2jBbFzg) to get a demo of the game and inspiration for how you might make changes using Copilot.
<img width="1006" alt="Screenshot 2024-09-13 at 11 25 48 AM" src="https://github.com/user-attachments/assets/ab85020f-ed20-43fc-a6ee-161cb61b3f69">

**Copilot resources:**
Using GitHub Copilot in your IDE: Tips, tricks, and best practices
 https://github.blog/developer-skills/github/how-to-use-github-copilot-in-your-ide-tips-tricks-and-best-practices/

Getting started with Copilot: https://docs.github.com/en/copilot/quickstart

**Comment by sudoyasir** [comment]
Excited to join the Copilot challenge! I'll be participating and looking forward to putting my own twist on the game. Can't wait to see what everyone comes up with!

**Comment by Ileolami** [comment]
I'm happy to be here.

**Comment by JuanGdev** [comment]
Happy to participate in this challenge! :octocat:

**Comment by iqnite** [comment]
Can I use this open-source music (with credits, of course)? Or are there copyright issues that might prevent my game from being rated? Here's the song: https://youtu.be/1RiaSfzxS90

**Reply by MetzinAround** [reply]
>💜 "Fire" is 👉FREE-to-use ﻿on YouTube, Twitch and other social media channels. HOWEVER TO PROTECT ﻿YOU FROM INVALID CLAIMS YOU WILL GET A CLAIM FROM MY MCN “THE DISTRICT”.﻿ You can dispute that claim, saying you have a license, and then simply ﻿put “TheFatRat” (without parentheses) into the note section. No further ﻿explanation needed. If you credit me properly (copy paste the credits ﻿below) the claim will be released and you can monetize the video ﻿yourself. Also you won’t lose any revenue that’s made between the dispute and the release. YouTube holds it back for you.

Looks like they said it's ok!

**Reply by iqnite** [reply]
> Looks like they said it's ok! 

Thank you for your reply!

**Comment by iqnite** [comment]
My game:
https://github.com/iqnite/CubeFlyer

**Reply by iqnite** [reply]
Updated!
Changelog: https://github.com/iqnite/CubeFlyer/releases
Play: https://github.com/iqnite/CubeFlyer

**Comment by sudoyasir** [comment]
Here's my version of game; [Astro Glide](https://github.com/sudoyasir/AstroGlide) 😀

**Comment by lu2000luk** [comment]
Here is my edition: https://github.com/lu2000luk/CubeFlyer
Probably going to make some changes to it in the next days but i submit it right now so i dont forget later 👀

**Reply by iqnite** [reply]
You definitely deserve one of those Mona prints!

**Reply by sudoyasir** [reply]
@iqnite No doubt bro 🔥

**Reply by lu2000luk** [reply]
> You definitely deserve one of those Mona prints!

🙏Thanks (=

**Comment by Ileolami** [comment]
[Here is my entry](https://github.com/Ileolami/CubeFlyer)

**Reply by iqnite** [reply]
Very cool design, love it!

**Comment by lu2000luk** [comment]
When will resoults be released? And also, the contest is over, i submitted my entry, can i still make new commits to it?

**Reply by iqnite** [reply]
There's some info [here](https://github.com/github-education-resources/Student-Developer-Pack-Current-Partners-FAQ/blob/main/Copilot-Game-Challenge-Terms-and-Conditions.md).

![image](https://github.com/user-attachments/assets/eb84560e-074b-4c82-b2c5-a273298478de)

About whether you're allowed to commit... I have no idea, but I hope so, as I've made some changes to the game after the deadline. I think that they'll count them in, as this is a contest and not an exam.

Again, you made a really cool game!

**Comment by morganersery1** [comment]
Thanks to all who participated! We had 15 forks of the repo and 4 final Discussion entries, which means: @iqnite, @lu2000luk, @Ileolami and @sudoyasir - you're all getting Mona prints! You should expect an email with a link to a form allowing you to provide an address where we can send your print. Congrats! And great work!

**Reply by Ileolami** [reply]
Wow

Thank you so much

**Reply by sudoyasir** [reply]
Thank you so much

**Reply by lu2000luk** [reply]
❤️

**Reply by iqnite** [reply]
Wow thanks a lot guys! Also for making this challenge, it kept me motivated to code!

**Reply by iqnite** [reply]
Did you guys receive your Mona print by now? Or are also waiting? Just to make sure I didn't do anything wrong.

**Reply by morganersery1** [reply]
Hello - sharing an update. All posters have been mailed out. They should arrive either by end of this week or end of next week via FedEx & DHL (depending on the recommended carrier for your area). Sorry for the delay. They're on their way!

**Reply by iqnite** [reply]
Got an SMS from FedEx this morning, though it was a scam at first. Thanks for the info and for the posters!

**Comment by Ileolami** [comment]
I didn't get any response ever since I've filled my details

**Reply by sudoyasir** [reply]
Same here

**Reply by morganersery1** [reply]
They're on the way! Had a little shipping set back but they'll be headed your way soon. You're not forgotten 😄

**Reply by iqnite** [reply]
Thanks for telling us, I thought that I'd inputted my details wrong! :)

**Reply by iqnite** [reply]
Hello, 
Just to let you know, today I received a letter from FedEx telling me that I have to pay **27€** of parcel customs for the "prize". Not only is this a slap in the face for us who worked on a project for this contest, but it's also against the terms and conditions.
I've downloaded all my private and public repos and am planning to host them **locally**. Won't trust a company of this kind anymore. I'm also **cancelling** my active GitHub subscriptions, and I advise you all to do the same. 
I am really disappointed, GitHub. What you did is lazy and greedy. Scamming teenagers like this seems to be a common practice among Silicon Valley companies, I feel so stupid for not having learned from what Microsoft, Replit, etc. did in the past.

**Reply by morganersery1** [reply]
Hi @iqnite we certainly understand your frustration. We were told that customs would not have been an issue in this case, but it is possible that we did not have the most up to date information. All fees should have been routed back to us for shipping of the posters. We will contact FedEx to better understand the issue on our end, since, as you correctly stated, winners would not be charged as a part of the challenge. We hope to regain the trust that was lost by this experience and that this does not discourage you in your developer journey moving forward.

**Reply by iqnite** [reply]
Thanks for your care, and sorry for my rough words before. I know mistakes can happen, but as a few weeks ago another company left its customers (including me) behind, my distrust into tech companies has grown. GitHub's ecosystem (Git, VSCode, Copilot, ...) is my entire workflow, and I honestly don't know what I would do without them. Again, thanks for your clarification. So, just to be on the safe side, I don't have to pay for anything now, right?

**Reply by morganersery1** [reply]
Thank you for your understanding. Yes, please allow us to look into this. Customs fees are sometimes imposed by the recipient's custom agency. We sometimes have problems shipping swag and it looks like it may be the case here. We will contact FedEx to see how we can resolve. In some cases the sender is able to pay the custom fee on behalf of the recipient.

**Reply by sudoyasir** [reply]
Same thing happened with me @morganersery1

**Reply by morganersery1** [reply]
@iqnite - Update for you is that I was able to get in touch with FedEx. They've sent a request into the accepting office's representatives and I should have an update on what we need to do to resolve in ~1 day. @sudoyasir I've found that yours was fully returned, but this means that we can just resend it. I'm checking now to see if we can pay the customs fees in advance so that it can be delivered on this second try. I will update you both further via email since I don't want your sensitive shipping information on a public thread. But know that I am working to resolve both.

**Reply by iqnite** [reply]
Hello, I got the same letter again, this time saying that if I don't pay within 7 days, I get into legal trouble...

**Reply by iqnite** [reply]
@morganersery1

**Reply by morganersery1** [reply]
@iqnite Is there a way to share who you're in contact with and or have the package released for return so that we can resolve?

**Reply by morganersery1** [reply]
@iqnite I was able to get in touch with Fedex again regarding this shipment. They say it was delivered according to your local post, but since you're saying it was not, I am filing a claim to ensure they continue investigating.

**Reply by iqnite** [reply]
> @iqnite Is there a way to share who you're in contact with and or have the package released for return so that we can resolve?
> 

What do you mean?

**Reply by iqnite** [reply]
> @iqnite I was able to get in touch with Fedex again regarding this shipment. They say it was delivered according to your local post, but since you're saying it was not, I am filing a claim to ensure they continue investigating.

It's not a problem with the delivery, the problem is the payment of the fees. Or did I misunderstand something?

**Reply by morganersery1** [reply]
@iqnite I'm asking for clarification if they've delivered it and are asking for fees or if they're withholding the shipment (and will not deliver) until the fees are paid. I believe both can be resolved with certain paperwork but I would need to know which is the case. If it's request for customs fees has been made, it should be from a local customs office in your country, which is why I ask for a contact. If it is easier, you can forward me the letter you received so that we can work to resolve.

**Reply by iqnite** [reply]
> @iqnite I'm asking for clarification if they've delivered it and are asking for fees or if they're withholding the shipment (and will not deliver) until the fees are paid. 

I have already received the package, and there's no way to return it. I will email you the letter as soon as possible.

**Reply by lu2000luk** [reply]
My package hasent arrived yet.

**Reply by sudoyasir** [reply]
Same for me :(

**Comment by Ileolami** [comment]
Awesome, thanks for the update!

**Comment by Ileolami** [comment]
I've received my gift

Thank you so much
